### PR TITLE
Resolves #848: Fixing module resolution with nonstandard filenames

### LINKED ISSUE #848: Regression in 2.14.0
```
git clone https://github.com/vaadin/flow.git
cd flow
mvn versions:set -DnewVersion=1.2.3
```

Expected outcome:
All modules are set to version 1.2.3, like with 2.13.0

Actual outcome:
```
[ERROR] 
java.nio.file.NoSuchFileException: /.../flow/flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom.xml
    at sun.nio.fs.UnixException.translateToIOException (UnixException.java:92)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:106)
    at sun.nio.fs.UnixException.rethrowAsIOException (UnixException.java:111)
    at sun.nio.fs.UnixFileSystemProvider.newByteChannel (UnixFileSystemProvider.java:219)
    at java.nio.file.Files.newByteChannel (Files.java:380)
    at java.nio.file.Files.newByteChannel (Files.java:432)
    at java.nio.file.spi.FileSystemProvider.newInputStream (FileSystemProvider.java:422)
    at java.nio.file.Files.newInputStream (Files.java:160)
    at org.codehaus.plexus.util.xml.XmlReader.<init> (XmlReader.java:129)
    at org.codehaus.plexus.util.xml.XmlStreamReader.<init> (XmlStreamReader.java:67)
    at org.codehaus.plexus.util.ReaderFactory.newXmlReader (ReaderFactory.java:122)
    at org.codehaus.mojo.versions.api.PomHelper.readXmlFile (PomHelper.java:1498)
    at org.codehaus.mojo.versions.AbstractVersionsUpdaterMojo.process (AbstractVersionsUpdaterMojo.java:326)
    at org.codehaus.mojo.versions.SetMojo.execute (SetMojo.java:381)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:370)
```

and the version in the following modules is not updated:
```
flow-tests/test-ccdm-flow-navigation/pom-production.xml
flow-tests/test-ccdm/pom-production.xml
flow-tests/test-custom-frontend-directory/test-themes-custom-frontend-directory/pom-generatedTsDir.xml
flow-tests/test-frontend/test-npm/pom-production.xml
flow-tests/test-frontend/test-pnpm/pom-production.xml
flow-tests/test-frontend/vite-context-path/pom-production.xml
flow-tests/test-frontend/vite-embedded/pom-production.xml
flow-tests/test-frontend/vite-pwa-custom-offline-path/pom-production.xml
flow-tests/test-frontend/vite-pwa-disabled-offline/pom-production.xml
flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-devmode.xml
flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-prod-fallback.xml
flow-tests/test-npm-only-features/test-npm-bytecode-scanning/pom-production.xml
flow-tests/test-pwa-disabled-offline/pom-production.xml
flow-tests/test-pwa/pom-production.xml
```

### PR DESCRIPTION
**Pull request by ajarmoniuk**
@slawekjaranowski 

Please review.

### CODE REVIEW

File: versions-maven-plugin/src/it/it-set-issue-848/pom.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>default-group</groupId>
+    <artifactId>default-artifact</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>moduleA/moduleA.xml</module>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-set-issue-848/pom.xml:11]: Please add something  like:

```
<module>moduleA/moduleA1.xml</module>
<module>moduleA/moduleA2.xml</module>
```

Comment (slawekjaranowski) [reply versions-maven-plugin/src/it/it-set-issue-848/pom.xml:11]: Fix for it should be in `org.codehaus.mojo.versions.api.PomHelper#getReactorModels`

We have:
```
String modulePath = (!path.isEmpty() && !path.endsWith("/") ? path + "/" : path)
                            + pomFile.getParentFile().getName();
```

So we drop info info about pom name in the same directory.

Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-set-issue-848/pom.xml:11]: Good point! Thanks.

**Comment by ajarmoniuk** [comment]
A very valid remark, applied the necessary fix. Thanks again.

File: versions-common/src/main/java/org/codehaus/mojo/versions/api/PomHelper.java [hunk 2]
```diff
@@ -1362,48 +1334,39 @@ public static ProjectBuildingRequest createProjectBuilderRequest(
...
                         // the aim of this goal is to fix problems when the project cannot be parsed by Maven,
                         // so we have to work with the raw model and not the interpolated parsed model from maven
                         Model moduleModel = getRawModel(pomFile);
-                        result.put(modulePath, moduleModel);
-                        childResults.putAll(getReactorModels(modulePath, moduleModel, project, logger));
+                        result.put(pomFile, moduleModel);
+                        childResults.putAll(getChildModels(moduleModel, project, logger));
                     } catch (IOException e) {
                         logger.debug("Could not parse " + pomFile.getPath(), e);
                     }
```
Comment (slawekjaranowski) [inline versions-common/src/main/java/org/codehaus/mojo/versions/api/PomHelper.java:1372]: At least should be an error, maybe we should not cache it.
Now with a problem - build status is ok - but not every module are updated.

File: versions-maven-plugin/src/it/it-set-issue-848/pom.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    <groupId>default-group</groupId>
+    <artifactId>default-artifact</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>pom</packaging>
+
+    <modules>
+        <module>moduleA/moduleA1.xml</module>
+        <module>moduleA/moduleA2.xml</module>
+        <module>moduleB/pom.xml</module>
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-set-issue-848/pom.xml:13]: please add one more case from issue with 

```
<relativePath>../..</relativePath>
```

https://github.com/mojohaus/versions/issues/848#issuecomment-1351734726

File: versions-common/src/main/java/org/codehaus/mojo/versions/api/PomHelper.java (hunk 2 above)
Comment (ajarmoniuk) [reply versions-common/src/main/java/org/codehaus/mojo/versions/api/PomHelper.java:1372]: Well, I did not want to change the existing behaviour. Should we then introduce a toggle for that? Or straight on throw an error?

Comment (ajarmoniuk) [reply versions-common/src/main/java/org/codehaus/mojo/versions/api/PomHelper.java]: @slawekjaranowski applied the suggested change, please advise on the above

**Comment by ajarmoniuk** [comment]
Forgot to check the `site` goal 🤦

**Pull request merged**